File: examples/nested.rs
use taffy::prelude::*;

fn main() -> Result<(), taffy::error::TaffyError> {
    let mut taffy = Taffy::new();

    // left
    let child_t1 = taffy
        .new_leaf(FlexboxLayout { size: Size::<Option<Dimension>>::from_points(5.0, 5.0), ..Default::default() })?;

    let div1 = taffy.new_with_children(
        FlexboxLayout {
            size: Size::<Option<Dimension>>::from_percent(0.5, 1.0),
            // justify_content: JustifyContent::Center,
            ..Default::default()
        },
        &[child_t1],
    )?;

    // right
    let child_t2 = taffy
        .new_leaf(FlexboxLayout { size: Size::<Option<Dimension>>::from_points(5.0, 5.0), ..Default::default() })?;

    let div2 = taffy.new_with_children(
        FlexboxLayout {
            size: Size::<Option<Dimension>>::from_percent(0.5, 1.0),
            // justify_content: JustifyContent::Center,
            ..Default::default()
        },
        &[child_t2],
    )?;

    let container = taffy.new_with_children(
        FlexboxLayout { size: Size::<Option<Dimension>>::from_percent(1.0, 1.0), ..Default::default() },
        &[div1, div2],
    )?;

    taffy.compute_layout(container, Size { height: Some(100.0), width: Some(100.0) })?;

    println!("node: {:#?}", taffy.layout(container)?);

    println!("div1: {:#?}", taffy.layout(div1)?);
    println!("div2: {:#?}", taffy.layout(div2)?);

    println!("child1: {:#?}", taffy.layout(child_t1)?);
    println!("child2: {:#?}", taffy.layout(child_t2)?);

    Ok(())
}

File: bindings/java/java/examples/java/com/dioxuslabs/taffy/Nested.java
package com.dioxuslabs.taffy;

import com.dioxuslabs.taffy.geom.Size;
import com.dioxuslabs.taffy.geom.measure.Dimension;
import com.dioxuslabs.taffy.style.Style;

import java.util.List;

/**
 * Equivalent of taffy/examples/nested.rs
 */
public class Nested {
    public static void main(String[] args) {
        TaffyTree taffy = new TaffyTree();

        // left
        long childT1 = taffy.newLeaf(
                Style.builder().size(Size.lengthDimension(5, 5))
        );

        long div1 = taffy.newWithChildren(
                Style.builder().size(new Size<>(Dimension.percent(0.5f), Dimension.percent(1f))),
                List.of(childT1)
        );

        // right
        long childT2 = taffy.newLeaf(
                Style.builder().size(Size.lengthDimension(5, 5))
        );

        long div2 = taffy.newWithChildren(
                Style.builder().size(new Size<>(Dimension.percent(0.5f), Dimension.percent(1f))),
                List.of(childT2)
        );

        long container = taffy.newWithChildren(
                Style.builder().size(Size.percent(Dimension.class, 1, 1)),
                List.of(div1, div2)
        );

        taffy.computeLayout(
                container,
                Size.definiteAvailableSize(100, 100)
        );
        System.out.println("node: " + taffy.layout(container).toString());

        System.out.println("div1: " + taffy.layout(div1).toString());
        System.out.println("div2: " + taffy.layout(div2).toString());

        System.out.println("child1: " + taffy.layout(childT1).toString());
        System.out.println("child: " + taffy.layout(childT2).toString());
    }
}
